src/main/java/org/jabref/model/entry/Month.java

```diff
@@ -93,9 +95,31 @@
             int number = Integer.parseInt(value);
             return Month.getMonthByNumber(number);
         } catch (NumberFormatException e) {
-            return Optional.empty();
+            return parseGermanShortMonth(testString);
```

Comment:
Better move the method above in the parse method, so the parseGermanMonth will be executed when the others returned empty. Just like it it is deone with the getMonthByShortName

Reply:
Sorry, I don't know exactly what you mean.

Reply:
Okay, I will try to explain:
Move the method parseGermanShortMonth before the catch block. and after the getMonthByShortName:
```
 month= Month.parseGerman....
if(month.isPresent)
return month;
```
The end result is the same, but it looks more logical in the whole parse method:
1. We check if the value is empty
2. We try to get the normal short month
3. we try german short month
4. last try:  try into parse it as number 

Otherwise from a quick look I would see and think: Okay, if the integer parsing fails it implies it is automatically a German month. And that might not be the case. In reality the integer parsing maybe failed because we entered 9999

Reply:
Thanks for the detailed explanation. I adjusted it.